internal/archive/archive.go

```diff
@@ -394,7 +394,9 @@ func (index *ubuntuIndex) fetch(suffix, digest string, flags fetchFlags) (io.Rea
 	switch resp.StatusCode {
 	case 200:
 		// ok
-	case 401, 404:
+	case 401:
+		return nil, fmt.Errorf("cannot fetch from %s: authorization required", index.archive.baseURL)
```

Comment:
Small comment but I think it is better to say `"cannot fetch from %s: unauthorized"` because authorization required could be interpreted as you did not supply token, when it could be that the token is incorrect. Additionally "Unauthorized" is also the name of HTTP 401 response, so it is more fitting.

I also don't think we should use the baseURL for this. If a user sees the URL he has to know the mapping from URL to pro value and then to archive, which is simple enough given that the URL contains "fips"/"esm"/whatever, but it would be better if we referred to the archive name instead. That way your could see the error and know what archive in chisel.yaml it corresponds to.

Reply:
EDIT: Just saw that you already changed the code to use the label instead of the url. Nice!

Reply:
The bit about the error message and "unathorized" still applies :)

Reply:
Updated in 334d55a808a1cf05cc01a3f365a7387fcfc04da8. :)